cli/src/progress.rs

```diff
@@ -61,7 +61,15 @@ impl Progress {
             _ = write!(output, "{}", crossterm::cursor::Hide);
             self.guard = Some(guard);
         }
-        self.next_print = now.min(self.next_print + Duration::from_secs(1) / UPDATE_HZ);
+        let period = Duration::from_secs(1) / UPDATE_HZ;
+        // Try to print periodically by setting the new threshold exactly one period
+        // later
+        self.next_print = self.next_print + period;
+        // But if it's been long enough since last update that we'd effectively print
+        // two updates now, reset the threshold to now.
+        if now >= self.next_print {
+            self.next_print = now + period;
```

Comment:
Might be simpler to just do this unconditionally? Means more drift, but it's not really important that we exactly nail the update rate.

Reply:
I'm fine with simplifying it to `self.next_print = now + period;`

Reply:
Done. I also moved this line up a bit to be closer to the `if now < self.next_print {` line.